### ISSUE TITLE
Job stuck InProgress status when synchronizing a folder with no changes

### ISSUE THREAD

**Issue by alloro** [Azure/azure-storage-azcopy#826]
### Which version of the AzCopy was used? 
10.3.3

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux

### What command did you run?
`azcopy sync '<local folder>' '<blob uri w/sas>'`

### What problem was encountered?
The job was always in a state of **InProgress** even tho it exited with the message `The source and destination are already in sync`

### How can we reproduce the problem in the simplest way?
Try synchronizing a local folder that does not change with a blob that does contain the same data as the local folder.

The job only changes the state to **Completed** when you add a file in the local folder and then fire up the sync command again. The job that was previously fired is stuck **InProgress** and will never change its status. 

### Have you found a mitigation/solution?
Nope

**Comment by JohnRusk** [comment]
Thanks for logging this.  I'm currently investigating a log file from another report of a similar issue.  Not sure what the cause is yet, because I've only just started investigating.

**Comment by JohnRusk** [comment]
May be duplicate of https://github.com/microsoft/AzureStorageExplorer/issues/2515

If you could send me the log files and plan files for the job (as per notes in https://github.com/microsoft/AzureStorageExplorer/issues/2515) I might be able to confirm whether this is or isn't the same issue.

**Comment by JohnRusk** [comment]
BTW log files (and plan files) contain filenames but no credentials.

**Comment by alloro** [comment]
This is the content of the log file: 
> 2020/01/14 09:16:28 AzcopyVersion  10.3.3
2020/01/14 09:16:28 OS-Environment  linux
2020/01/14 09:16:28 OS-Architecture  amd64
2020/01/14 09:16:28 Log times are in UTC. Local time is 14 Jan 2020 09:16:28
2020/01/14 09:16:28 Job-Command sync /opt/app/data https://<url w/sas>
2020/01/14 09:16:28 Number of CPUs: 8
2020/01/14 09:16:28 Max file buffer RAM 4.000 GB
2020/01/14 09:16:28 Max concurrent network operations: 128 (Based on number of CPUs. Set AZCOPY_CONCURRENCY_VALUE environment variable to override)
2020/01/14 09:16:28 Check CPU usage when dynamically tuning concurrency: true (Based on hard-coded default. Set AZCOPY_TUNE_TO_CPU environment variable to true or false override)
2020/01/14 09:16:28 Max concurrent transfer initiation routines: 64 (Based on hard-coded default. Set AZCOPY_CONCURRENT_FILES environment variable to override)
2020/01/14 09:16:28 Max open files when downloading: 1048134 (auto-computed)
2020/01/14 09:16:28 ERROR: No transfers were scheduled.

I had to redact the url because it had a sas. I also checked the plan file with `xxd` and it seemed that it holds the same information (run command).

I also tried running the `azcopy` binary that I build from your PR #828, and the result was the same as before. I'm not sure if this PR should have fixed it.

**Comment by JohnRusk** [comment]
Thanks for the clarification. This is a different problem from the other one I mentioned above.

Are you using the reported job status in a script?  If not, can you explain a bit about what you need the status for, and I may be able to suggest a workaround.

**Comment by alloro** [comment]
I am using the `azcopy sync` as a cronjob to sync metadata of an application to azure blob storage. 

That being said, the azcopy creates a new plan (which is ~8KB in size) for every run. Due to the fact that the job does not transfer any data in the case when everything is already in sync the job status results in **InProgress** even tho the job finished and told me **Everything in sync**. 

Now that I have a lot of jobs **InProgress** I cannot judge if they are still running or if they did really complete. Meaning that I cannot cleanup jobs with `azcopy jobs clean --with-status=...` and therefore I accumulate quite a lot of space on my disk just by having those zombie jobs that actually finished but are just still in a **InProgress** state.

If the status would change correctly to **Completed** I could do another cronjob to clean **Completed** jobs within a certain timeframe/interval.


TL;DR: I run `azcopy sync` within a cronjob to sync metadata to azblobstorage. The jobs that azcopy creates finish with the message `Already in Sync source/destination` but job remains in status **InProgress** even tho its finished and should be in status **Completed**. Therefore the folder `/root/.azcopy/plans` gets polluted. I cannot clean the jobs because I don't know if they are actually running -- I am not watching the output of azcopy due to the cronjob.

**Comment by JohnRusk** [comment]
Thanks. That's a really clear description, thanks. The workaround I was going to suggest was to watch the output, but as you say that's not workable in your case.

@zezha-msft Any other workaround ideas?

**Comment by alloro** [comment]
Hi 👋, 
is there any progress on this issue?

**Comment by JohnRusk** [comment]
No, sorry. It's not scheduled for our up-coming 10.4 release.  Is there any chance you can use the exit code of the application to help you?  It returns a standard exit code to the OS.

**Comment by marcelovvm** [comment]
Hi, I have the same issue here. The azcopy sync command when executed on a set of resources already synchronized, ends the process with the message: The source and destination are already in sync. And when I go to check the status of the job... it is: Final Job Status: InProgress. I will still assemble my PS scripts to automate the synchronization of several resources, so how can I know if the jobs really finished successfully (= synchronized everything that had to be synchronized). Live long and prosper!

**Comment by JohnRusk** [comment]
As a possible workaround, maybe have a look at the application's exit code too.  I can't remember, but maybe it gives a "success" exit code in the case you're looking at.

**Comment by marcelovvm** [comment]
$? = True... could be like that?

**Comment by JohnRusk** [comment]
I can't remember the Linux syntax sorry.   To test, run a transfer of something that you know will work, and something that you know will fail (e.g. source directory doesn't exist).  That will show you the two possible values of the exit code.

**Comment by marcelovvm** [comment]
$? its a PowerShell syntax... it returns code of the last command executed.

**Comment by mohsha-msft** [comment]
Closing due to age of issue. The service and the tool have both evolved quite a bit. Please let us know if you still experience this.
